stream.go

```diff
@@ -0,0 +1,355 @@
+	Expiry             *time.Time `json:"expiry,omitempty"`
+	UploadVideoOptions
+}
+
+// UploadVideoURLWatermark represents UID of an existing watermark.
+type UploadVideoURLWatermark struct {
+	UID string `json:"uid,omitempty"`
+}
+
+// StreamVideoCreate represents parameters returned after creating a video.
+type StreamVideoCreate struct {
+	UploadURL string               `json:"uploadURL,omitempty"`
+	UID       string               `json:"uid,omitempty"`
+	Watermark StreamVideoWatermark `json:"watermark,omitempty"`
+}
+
+// StreamListOptions represents parameters used when listing stream videos.
+type StreamListOptions struct {
+	After         *time.Time `url:"after,omitempty"`
+	Before        *time.Time `url:"before,omitempty"`
+	Creator       string     `url:"creator,omitempty"`
+	IncludeCounts bool       `url:"include_counts,omitempty"`
+	Search        string     `url:"search,omitempty"`
+	Limit         int        `url:"limit,omitempty"`
+	Asc           bool       `url:"asc,omitempty"`
+	Status        string     `url:"status,omitempty"`
+}
+
+// StreamSignedURLOptions represent parameters used when creating a signed URL.
+type StreamSignedURLOptions struct {
+	ID           string            `json:"id,omitempty"`
+	PEM          string            `json:"pem,omitempty"`
+	EXP          int               `json:"exp,omitempty"`
+	NBF          int               `json:"nbf,omitempty"`
+	Downloadable bool              `json:"downloadable,omitempty"`
+	AccessRules  StreamAccessRules `json:"accessRules,omitempty"`
+}
+
+// StreamVideoResponse represents an API response of a stream video.
+type StreamVideoResponse struct {
+	Response
+	Result StreamVideo `json:"result,omitempty"`
+}
+
+// StreamVideoCreateResponse represents an API response of creating a stream video.
+type StreamVideoCreateResponse struct {
+	Response
+	Result StreamVideoCreate `json:"result,omitempty"`
+}
+
+// StreamListResponse represents the API response from a StreamListRequest.
+type StreamListResponse struct {
+	Response
+	Result []StreamVideo `json:"result,omitempty"`
+	Total  string        `json:"total,omitempty"`
+	Range  string        `json:"range,omitempty"`
+}
+
+// StreamSignedURLResponse represents an API response for a signed URL.
+type StreamSignedURLResponse struct {
+	Response
+	Result struct {
+		Token string `json:"token,omitempty"`
+	}
+}
+
+// StreamAccessRules represents the accessRules when creating a signed URL
+//I feel like should be implemented somewhere already but couldn't find it
+// accessRules under https://api.cloudflare.com/#stream-videos-create-a-signed-url-token-for-a-video
+type StreamAccessRules []map[string]interface{}
+
+// StreamUploadFromURL send a video URL to it will be downloaded and made available on Stream.
+//
+// API Reference: https://api.cloudflare.com/#stream-videos-upload-a-video-from-a-url
+func (api *API) StreamUploadFromURL(ctx context.Context, accountID string, options StreamUploadFromURLOptions) (StreamVideo, error) {
```

Comment:
let's move accountID into `StreamUploadFromURLOptions`. allows us more flexibility in the future if the product is available in other contexts (user, zone, zoneless)